matsim/src/main/java/org/matsim/core/router/NetworkRoutingInclAccessEgressModule.java

```
@@ -98,144 +101,166 @@
-		if ( lastPlanElement instanceof Leg ) {
-			final Activity interactionActivity = createInteractionActivity( startCoord, startLinkId, stageActivityType );
-			result.add( interactionActivity ) ;
+		PlanElement lastPlanElement = result.get(result.size() - 1);
+		if (lastPlanElement instanceof Leg) {
+			final Activity interactionActivity = createInteractionActivity(startCoord, startLinkId, stageActivityType);
+			result.add(interactionActivity);
 		} else {
 			// don't add another (interaction) activity
 			// TODO: assuming that this is an interaction activity, e.g. walk - drt interaction - walk
 			// Not clear what we should do if it is not an interaction activity (and how that could happen).
 		}
 
 		Id<Link> endLinkId = toFacility.getLinkId();
-		if ( endLinkId==null ) {
+		if (endLinkId == null) {
 			endLinkId = startLinkId;
 		}
 
 		if (mode.equals(TransportMode.walk)) {
-			Leg egressLeg = populationFactory.createLeg( TransportMode.non_network_walk ) ;
-			egressLeg.setDepartureTime( now );
-		routeBushwhackingLeg(person, egressLeg, startCoord, toFacility.getCoord(), now, startLinkId, endLinkId, populationFactory, config ) ;
-			result.add( egressLeg ) ;
+			Leg egressLeg = populationFactory.createLeg(TransportMode.non_network_walk);
+			egressLeg.setDepartureTime(now);
+			routeBushwhackingLeg(person, egressLeg, startCoord, toFacility.getCoord(), now, startLinkId, endLinkId, populationFactory, config);
+			result.add(egressLeg);
+		} else if (config.plansCalcRoute().getAccessEgressWalkType().equals(AccessEgressWalkType.readAccessTimeFromLinkAttribute)) {
+			Leg egressLeg = populationFactory.createLeg(TransportMode.walk);
+			egressLeg.setDepartureTime(now);
+			routeBushwhackingLeg(person, egressLeg, startCoord, toFacility.getCoord(), now, startLinkId, endLinkId, populationFactory, config);
+			double accessTime = NetworkUtils.getLinkAccessTime(filteredNetwork.getLinks().get(endLinkId), mode).orElse(0.0);
+			egressLeg.setTravelTime(accessTime);
```

this is a dangerous fallback to 0s access time. Dangerous because you won't notice if no access time link attribute is found for the mode (it neither fails nor gives a warning).

---

Allright, let's throw an exception instead :)